app/services/invite_provider_user.rb

```diff
@@ -0,0 +1,60 @@
+require 'jwt'
+
+class InviteProviderUser
+  def initialize(provider_user_form:)
```

Comment:
I'm not sure about passing in the form here. Perhaps we can only pass in the `ProviderUser` object?

Reply:
In this iteration, we don't yet have `first_name`, `last_name` for the provider user. The current plan is to add this migration as a separate PR, together with code matching on uid when people sign-in to reconcile their email address, first name and last name. This way DfE Sign-In remains the source of truth for email and names.

Reply:
On one hand I agree the less coupling the better, so `form <-> model` and `service <-> model` interactions would be kept separate by adding temporary attributes to the model, however a `service <-> form` interaction has the advantage the service can add API errors directly to the form. Not sure what the right way is here. I don't think `service <-> form <-> model` is too bad.

Reply:
Progress. See comments on controller above. We'll have `service <-> model`, but the model will be built by the form. The controller will be responsible for communicating any errors to the appropriate objects.